lib/src/install.rs

```diff
@@ -925,6 +925,38 @@ pub(crate) fn setup_tmp_mounts() -> Result<()> {
     Ok(())
 }
 
+#[context("Ensuring sys mounts")]
+pub(crate) fn setup_sys_mounts() -> Result<()> {
+    tracing::debug!("Setting up sys mounts");
+    // First of all, does efivars even exist in the host? If not, we are
+    // not dealing with an EFI system
+    let _ = rustix::fs::lstat("/proc/1/root/sys/firmware/efi/efivars")?;
+
+    // Now, let's find out if it's populated
+    let od = rustix::fs::open(
+        "/proc/1/root/sys/firmware/efi/efivars",
+        rustix::fs::OFlags::DIRECTORY,
+        rustix::fs::Mode::empty(),
+    )?;
+
+    let dir_populated = rustix::fs::Dir::read_from(od)?.next().is_some();
+    if dir_populated {
```

Comment:
Combining the above I think something just like:

```
let efivars = "/proc/1/root/sys/firmware/efi/efivars";
if !Path::new(efivars).try_exists()? {
    return Ok(());
}
if std::fs::read_dir(efivars)?.next().is_none() {
    return Ok(());
}
...rest of code
```
or so?

Reply:
Another option is https://docs.rs/cap-std-ext/latest/cap_std_ext/dirext/trait.CapStdExtDirExt.html#tymethod.open_dir_optional which we use in a few places in this project, and is intended for this use case.